converter/src/main/java/org/dswarm/converter/schema/XMLSchemaParser.java

```diff
@@ -426,6 +531,25 @@ private void addRDFValueAttributePath(final boolean addRDFValueAttributePath, fi
 		return Optional.fromNullable(schema);
 	}
 
+	private Optional<Schema> createSchema(final String uuid, final String name) throws DMPPersistenceException {
+
+		final Optional<Schema> optionalSchema = createSchema(uuid);
+
+		if (!optionalSchema.isPresent()) {
+
+			return Optional.absent();
+		}
+
+		final Schema schema = optionalSchema.get();
+
+		if (name != null) {
+
+			schema.setName(name);
+		}
+
+		return Optional.of(schema);
+	}
```

Comment:
basically, this (ifPresent/get) is exactly what the `map` combinator is for (called `transform` in guava):

```java
		return optionalSchema.transform(schema -> {
			if (name != null) {
				schema.setName(name);
			}
			return schema;
		});
```

It's usable with java8, although I'd say it makes more sense to use j.u.Optinal instead.